### PR TITLE
Death of mutant bodyparts AND external organs

### PR DESCRIPTION
**Pull request by Time-Green**
## About The Pull Request

Removes mutant bodyparts and external organs from the game completely
Digitgrade behaviour was mutant bodypart for no reason

Cat ears now work with the bodyparts overlay system, same as all the other external organs (since all their behaviour is now just on /organ

It doesn't remove all the /external types, but moves all behaviour to /organ. I'll follow up with a PR wiping all the /external organ types, but it's just conflict heaven so not this PR

I've also streamlined a lot of duplicate/weird species regeneration code

Melbert did the same PR as well but due to a lack of time (?) I have absorbed his PR to double nuke mutant bodyparts

## Why It's Good For The Game
Frees us from the chain of unmodular code, and kills my greatest nemesis (after the shuttle meteor murder bug)

## Changelog
:cl: Time-Green and MrMelbert
Refactor: External organ behaviour has been moved to /organ, ears now use the same system as the other organs
Refactor: Mutant bodyparts are dead! This likely does not mean much to the average person but it's very dear to me
code: Improves digitgrade handling in preference code
/:cl:

I have absorbed #85126, using Melberts code to improve and add some missing changes. Mainly improving the functioning of preferences and digitgrade legs. I didn't take over the hairstyle improvements.

### CODE REVIEW

**Comment by Time-Green** [comment]
@MrMelbert lol

**Comment by lessthnthree** [comment]
hold on to your mutant butts!

**Comment by Jolly-66** [comment]
Downstreams in shambles rn

**Comment by Time-Green** [comment]
> Downstreams in shambles rn

I made this a month ago and then forgot but right fuck I was gonna add a disclaimer that this is downstream cbt (as usual)

**Comment by ZephyrTFA** [comment]
It may be CBT but it also makes bugs and integration easier for the future

	

	

	

aka cope

File: code/modules/surgery/organs/internal/ears/_ears.dm [hunk 1]
```diff
@@ -136,28 +135,34 @@
...
-/obj/item/organ/internal/ears/cat/on_mob_remove(mob/living/carbon/human/ear_owner)
-	. = ..()
-	if(istype(ear_owner) && ear_owner.dna)
-		color = ear_owner.hair_color
-		ear_owner.dna.species.mutant_bodyparts -= "ears"
-		ear_owner.update_body()
+	bodypart_overlay = /datum/bodypart_overlay/mutant/cat_ears
+
+/// Bodypart overlay for the mushroom cap organ
+/datum/bodypart_overlay/mutant/cat_ears
```
Comment (OrionTheFox) [inline code/modules/surgery/organs/internal/ears/_ears.dm:146]: I was just scrolling through out of morbid curiosity and noticed this, should probably point it out

The comment says mushroom cap organ on cat ears

File: code/modules/unit_tests/organ_set_bonus.dm [hunk 2]
```diff
@@ -30,7 +30,7 @@
 		// Attempt to insert entire list of mutant organs for the given infusion_entry.
 		for(var/obj/item/organ/organ as anything in output_organs)
 			organ = new organ()
-			TEST_ASSERT(organ.Insert(lab_rat, special = TRUE, movement_flags = DELETE_IF_REPLACED), "The organ `[organ.type]` for `[infuser_entry.type]` was not inserted in the mob when expected, Insert() returned falsy when TRUE was expected.")
```
Comment (Time-Green) [inline code/modules/unit_tests/organ_set_bonus.dm:33]: This never did anything as Insert ALWAYS returns TRUE (which I've removed cause it's dumb

**Comment by Time-Green** [comment]
Probable screenshot test error in 10 minutes? I'm tired of cat ear bullshit I'll let Melberts fix deal with that :)

**Comment by Majkl-J** [comment]
As one of the downstream maintainers: I both love and hate you for this.

**Comment by StrangeWeirdKitten** [comment]
I'm shaking in my boots and crying in the corner. But then again, it forces us to refactor our shitty furry code :^)

**Comment by Time-Green** [comment]
Drafted until https://github.com/tgstation/tgstation/pull/85126 is merged

**Comment by Time-Green** [comment]
I talked to @MrMelbert and have absorbed his PR. Preferences now also work because Melbert is a way better coder than me. I've removed some code sections because the external organ removal that I packaged in gets rid of a lot of special code we needed to run.

I didn't move over the hairstyle changes. The PR is already pretty big (and not super modular) and it's a good change for an isolated PR

**Comment by Sadboysuss** [comment]
Looks like this is causing some issues https://github.com/tgstation/tgstation/issues/85321

**Comment by norsvenska** [comment]
this completely breaks organ manipulation surgery; inserting an organ sends it to nullspace.

**Comment by Time-Green** [comment]
I removed Insert always returning TRUE in favor of returning nothing, but some code depended on it that I didn't catch. Surgeries would insert organs, think it failed and send them to nullspace. Heretic sacrafices would also think the Inserts failed and kill the person in retaliation

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 3]
```diff
@@ -3,11 +3,10 @@
 	name = "Felinid"
 	id = SPECIES_FELINE
 	examine_limb_id = SPECIES_HUMAN
-	mutant_bodyparts = list("ears" = "Cat", "wings" = "None")
 	mutantbrain = /obj/item/organ/internal/brain/felinid
 	mutanttongue = /obj/item/organ/internal/tongue/cat
 	mutantears = /obj/item/organ/internal/ears/cat
```
Comment (lessthnthree) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm:8]:
```suggestion
```

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 4]
```diff
@@ -3,11 +3,10 @@
...
-	external_organs = list(
+	mutant_organs = list(
 		/obj/item/organ/external/tail/cat = "Cat",
```
Comment (lessthnthree) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm:10]:
```suggestion
		/obj/item/organ/external/tail/cat = "Cat",
		/obj/item/organ/internal/ears/cat = "Cat",
```

**Comment by Ghommie** [comment]
Could you check that this works with the recent changes, especially the digitigrade clothing by Melbert?

**Comment by Time-Green** [comment]
> Could you check that this works with the recent changes, especially the digitigrade clothing by Melbert?

@Ghommie Works well! I did the whole thingamajig (horribly mutilating lizards and putting them back toghether). Digitgrade clothes also work

**Pull request merged**